File: backend/models/grades.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class GradeEntryCreate(BaseModel):
    course_id: str
    assignment_title: str
    category: str
    score_earned: float
    score_possible: float
    due_date: Optional[datetime] = None
    source: str = "manual"  # manual | scraped | syllabus


class GoalRequest(BaseModel):
    course_id: str
    target_letter: str  # A, B+, B, etc.

File: frontend/types/grades.ts
export interface GradeEntry {
  id: string;
  courseId: string;
  assignmentTitle: string;
  category: string;
  scoreEarned: number;
  scorePossible: number;
  dueDate?: string | null;
  source: "manual" | "scraped" | "syllabus";
}

/** Maps to backend GradeEntryCreate for API calls */
export interface GradeEntryPayload {
  course_id: string;
  assignment_title: string;
  category: string;
  score_earned: number;
  score_possible: number;
  due_date?: string | null;
  source: string;
}

export interface ComputeGradeResponse {
  current_grade_pct: number;
  letter_grade: string;
  category_breakdown: Record<string, number>;
  weight_graded: number;
}

export interface GoalSimulatorResponse {
  required_pct: number | null;
  feasible: boolean;
  message: string;
}
